src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -151,6 +157,106 @@ private TableInfo buildLiveView(String bigQueryProject, String datasetName, Data
+        fieldList.add(Field.newBuilder("load_tag", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("load_time",  LegacySQLTypeName.TIMESTAMP)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("source_name", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("target_path", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("file_id", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("checksum_crc32c", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+        fieldList.add(Field.newBuilder("checksum_md5", LegacySQLTypeName.STRING)
+            .setMode(Field.Mode.REQUIRED)
+            .build());
+
+        return Schema.of(fieldList);
+    }
+
+    private static final String insertLoadHistoryToStagingTableTemplate =
+        "INSERT INTO `<project>.<dataset>.<stagingTable>`" +
+            " (load_tag, load_time, source_name, target_path, file_id, checksum_crc32c, checksum_md5)" +
+            " VALUES <load_history_array:{v|('<loadTag>', '<load_time>', '<v.sourcePath>', '<v.targetPath>'," +
+            " '<v.fileId>', '<v.checksumCRC>', '<v.checksumMD5>')}; separator=\",\">";
+
+    public void loadHistoryToStagingTable(
+        Dataset dataset,
+        String flightId,
+        String loadTag,
+        Instant loadTime,
+        List<BulkLoadHistoryModel> loadHistoryArray) throws InterruptedException {
+        BigQueryProject bigQueryProject = bigQueryProjectForDataset(dataset);
+
+        ST sqlTemplate = new ST(insertLoadHistoryToStagingTableTemplate);
+        sqlTemplate.add("project", bigQueryProject.getProjectId());
+        sqlTemplate.add("dataset", prefixName(dataset.getName()));
+        sqlTemplate.add("stagingTable", PDAO_LOAD_HISTORY_STAGING_TABLE_PREFIX + flightId);
+
+        sqlTemplate.add("load_history_array", loadHistoryArray);
+        sqlTemplate.add("load_tag", loadTag);
+        sqlTemplate.add("load_time", loadTime);
+
+        bigQueryProject.query(sqlTemplate.render());
+    }
+
+    private static final String mergeLoadHistoryStagingTableTemplate =
+        "MERGE `<project>.<dataset>.<loadTable>` L" +
+            " USING `<project>.<dataset>.<stagingTable>` S" +
+            " ON S.load_tag = L.load_tag AND S.load_time = L.load_time" +
+                " AND S.file_id = L.file_id" +
+            " WHEN NOT MATCHED THEN" +
+                " INSERT (load_tag, load_time, source_name, target_path, file_id, checksum_crc32c, checksum_md5)" +
+                " VALUES (load_tag, load_time, source_name, target_path, file_id, checksum_crc32c, checksum_md5)";
```

Comment:
@ddietterich  @fboulnois  Question: Should I also do a check here to makes sure that the staging table's data is the most recent? i.e. `WHEN NOT MATCHED AND S.load_time > L.load_time`

Reply:
I think we want an exact match. What we are trying to accomplish is not putting duplicate entries into the table.

While it _should_ be true that our load time is ahead of another load time, I get nervous about unsynchronized clocks, etc.